### DISCUSSION TITLE
HW 01, Problems Part 2, Question 3: Struggling to Understand Problem Statement

### DISCUSSION THREAD

**Discussion by feliciones** [cooperrc/computational-mechanics#26] [Q&A]
Hello,

I recently was working on the 3rd question and I wasn't sure if it was similar to the 2nd question, where A_99 would be the product of the indices, or in this case, of X and Y. Would someone be able to clarify?

Would it be incorrect for the mean to write a line like: `mean = (X+Y)/2` ?

**Comment by Aidan-Myatt** [comment]
Use the code Professor Cooper provided earlier to assign X and Y which is posted below. From that, assign the array A_99 using the array function which is also below.
X, Y = np.meshgrid(np.arange(10), np.arange(10))
A_99 = np.array((X, Y))

With these two lines of code you can find the mean and standard deviation of the A_99 array. You can do this with the np.mean() and np.std() functions, or you can you use another method to find it. Let me know if this works for you.

**Reply by feliciones** [reply]
Ahhh, thank you @Aidan-Myatt . Would I then need to import matplotlib for part d, or just write out the line: `contour([X, Y,] Z, [levels], **kwargs)`

**Reply by FishMacaron** [reply]
@feliciones I personally did that and formatted the code as "graph=plt.contourf(X,Y,A_99)"... but I'm not sure if that's the correct way to use confourf? I know this is a bit unrelated to the original post but I'm personally struggling a bit with how 3d is supposed to be coded since I'm not sure what the end product should look like. This is what my graph ended up looking like for reference:
![image](https://user-images.githubusercontent.com/68212585/171762622-a6c2dab8-fe52-49c2-a432-8cd3c3d653e9.png)

**Reply by Aidan-Myatt** [reply]
This is the code I used to plot it and the result. I am not 100% on it, but feel free to use it as a reference. 

![image](https://user-images.githubusercontent.com/106316906/171766890-c6bf79b3-5e35-4c56-be9e-dd3e0e20a1d2.png)
![image](https://user-images.githubusercontent.com/106316906/171766984-45182348-f4bd-4101-97ba-01cdb5cb7975.png)

**Reply by FishMacaron** [reply]
Okay, thank you! But is there a particular reason why you did that whole process for Z? (referencing *Z=(X\*\*2+Y\*\*2)\*\*0.5*) I think I understand part of the reason why is because otherwise Z wouldn't have an assigned value leading to an error, but is there any reason why you did that specifically?
What the contour function does, or its purpose of it, honestly confuses me so some simple things may be going over my head.

**Reply by cooperrc** [reply]
Here the goal is that `A_99` is the product of 0-to-9, e.g. 

```python
array([[ 0,  0,  0,  0,  0,  0,  0,  0,  0,  0],
       [ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9],
       [ 0,  2,  4,  6,  8, 10, 12, 14, 16, 18],
       [ 0,  3,  6,  9, 12, 15, 18, 21, 24, 27],
       [ 0,  4,  8, 12, 16, 20, 24, 28, 32, 36],
       [ 0,  5, 10, 15, 20, 25, 30, 35, 40, 45],
       [ 0,  6, 12, 18, 24, 30, 36, 42, 48, 54],
       [ 0,  7, 14, 21, 28, 35, 42, 49, 56, 63],
       [ 0,  8, 16, 24, 32, 40, 48, 56, 64, 72],
       [ 0,  9, 18, 27, 36, 45, 54, 63, 72, 81]])
```
